urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/Count.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * 
+ */
+package weibo4j;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import weibo4j.http.Response;
+import weibo4j.org.json.JSONArray;
+import weibo4j.org.json.JSONException;
+import weibo4j.org.json.JSONObject;
+
+/**
+ * @author SinaWeibo
+ *
+ */
+public class Count implements java.io.Serializable{
+
+	private static final long serialVersionUID = 9076424494907778181L;
+
+	private long id;
+	
+	private long comments;
+	
+	private long rt;
+	
+	private long dm;
+	
+	private long mentions;
+	
+	private long followers;
+	
+	public Count(JSONObject json)throws WeiboException, JSONException{
+    	id = json.getLong("id");
+    	comments = json.getLong("comments");
+    	rt = json.getLong("rt");
+    	dm = json.getLong("dm");
+    	mentions = json.getLong("mentions");
+    	followers = json.getLong("followers");
+    }
+	Count(Response res) throws WeiboException{
+	    JSONObject json=res.asJSONObject();
+	    try {
+		id = json.getLong("id");
+		comments = json.getLong("comments");
+	    	rt = json.getLong("rt");
+	    	dm = json.getLong("dm");
+	    	mentions = json.getLong("mentions");
+	    	followers = json.getLong("followers");
+	    
+	    } catch (JSONException je) {
+		throw new WeiboException(je.getMessage() + ":" + json.toString(),
+			je);
+	    }
+	    	
+	}
+	static List<Count> constructCounts(Response res) throws WeiboException {
+	   	 try {
+	   		 System.out.println(res.asString());
+	            JSONArray list = res.asJSONArray();
+	            int size = list.length();
+	            List<Count> counts = new ArrayList<Count>(size);
+	            for (int i = 0; i < size; i++) {
+	            	counts.add(new Count(list.getJSONObject(i)));
+	            }
+	            return counts;
+	        } catch (JSONException jsone) {
+	            throw new WeiboException(jsone);
+	        } catch (WeiboException te) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)